.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: ci
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: recursive
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - run: pip install --upgrade pip
+      - run: pip install numpy pexpect pytest setuptools
+      - run: pip install --editable modules/DroneCAN/pydronecan
+      - run: pip install --editable modules/mavlink/pymavlink
+      - run: pip install --editable Tools/ros2/ardupilot_dds_tests
+      - run: pytest --ignore=modules/gtest
+                    --ignore=modules/mavlink/pymavlink/generator/C/test/posix/sha256_test.py
+                    --ignore=modules/mavlink/pymavlink/tests/test_wp.py
+                    --ignore=modules/waf/playground
+                    --ignore=Tools/autotest/test_build_options.py
+                    --ignore=Tools/autotest/test_param_upgrade.py
+                    --ignore=Tools/FilterTestTool
+                    --ignore=Tools/ros2/ardupilot_dds_tests/test
```

Comment:
Why do we run the pip install for ardupilot_dds_tests, but then ignore them when running pytest?

Reply:
Because the following are not approved:
* #30444
* #30449
* #30487